code/crates/sync/src/state.rs

```diff
@@ -171,7 +176,9 @@ where
     /// When the tip height is higher than the requested range, then the request
     /// has been fully validated and it can be removed.
     pub fn remove_fully_validated_requests(&mut self) {
-        self.pending_requests
-            .retain(|_, (range, _)| range.end() > &self.tip_height);
+        self.pending_consensus_requests.retain(|_, (ranges, _)| {
+            ranges.retain(|range| range.end() > &self.tip_height);
+            !ranges.is_empty()
+        });
```

Comment:
note that, this will also update the `ranges` as well. is this what you want ?

did you mean?

```rs
        self.pending_consensus_requests.retain(|_, (ranges, _)| {
            let ranges = ranges.clone();
            ranges.retain(|range| range.end() > &self.tip_height);
            !ranges.is_empty()
        });
```

Reply:
Yes, we want to update the `ranges` here.